### PR TITLE
Use `glusterfsd --print-xlatordir`

### PR DESCRIPTION
**Pull request by prashanthpai**
Dependent change:
https://review.gluster.org/#/c/19905

Signed-off-by: Prashanth Pai <ppai@redhat.com>

### CODE REVIEW

File: glusterd2/xlator/load.go [hunk 1]
```diff
@@ -127,17 +129,14 @@ func loadXlator(xlPath string) (*Xlator, error) {
...
-	// glusterfs gets the path to xlator dir from a compile time flag named
-	// 'XLATORDIR' which gets passed through a -D flag to GCC. This isn't
-	// available to external programs via gluster CLI yet. When one or more
-	// versions of gluster are installed from source or otherwise, the
-	// following is the most fool-proof but hacky way to get the xlator dir
-	// location without making assumptions.
-
-	cmd := "strings -d `which glusterfsd` | awk '/glusterfs\\/.*\\/xlator$/'"
-	out, err := exec.Command("sh", "-c", cmd).Output()
+	out, err := exec.Command("gluster", "--print-xlatordir").Output()
```
Comment (atinmu) [inline glusterd2/xlator/load.go]: this brings in a hard dependency of shipping gluster cli always with GD2. I don't think its a good approach.

Comment (prashanthpai) [reply glusterd2/xlator/load.go]: Good catch!

Comment (aravindavk) [reply glusterd2/xlator/load.go]: Looks fine till this(https://github.com/gluster/glusterd2/issues/534) gets implemented

Comment (amarts) [reply glusterd2/xlator/load.go]: Good catch @atinmu but for atleast another 3 months, for users using/testing the feature, we should have an option to not crash. Should be very useful IMO.

File: glusterd2/xlator/load.go [hunk 2]
```diff
@@ -148,10 +147,10 @@ func getXlatorsDir() string {
 func loadAllXlators() (map[string]*Xlator, error) {
 
 	xlatorsDir := getXlatorsDir()
-
 	if xlatorsDir == "" {
 		return nil, fmt.Errorf("No xlators dir found")
 	}
+	log.WithField("xlatordir", xlatorsDir).Debug("Xlators dir found")
```
Comment (amarts) [inline glusterd2/xlator/load.go]: If you need to handle #666 then you should check for version < 4.0.0 here?

File: glusterd2/xlator/load.go (hunk 1 above)
Comment (atinmu) [reply glusterd2/xlator/load.go]: I'm OK to accept this patch as long as there's an issue filed to correct this later.

Comment (aravindavk) [reply glusterd2/xlator/load.go]: #534 captures these requirements.

**Comment by prashanthpai** [comment]
@atinmu @aravindavk I have moved the required options from gluster cli to `glusterfsd` binary: https://review.gluster.org/#/c/19905

File: glusterd2/xlator/load.go (hunk 2 above)
Comment (prashanthpai) [reply glusterd2/xlator/load.go]: That's been fixed in https://github.com/gluster/glusterd2/pull/667

**Comment by aravindavk** [comment]
> @atinmu @aravindavk I have moved the required options from gluster cli to glusterfsd binary: https://review.gluster.org/#/c/19905

Nice. Is it possible to add `--print-logdir` also to the `glusterfsd`? It is used in lot of places in glusterfs  regression tests.

**Comment by prashanthpai** [comment]
> Is it possible to add --print-logdir also to the glusterfsd? It is used in lot of places in glusterfs regression tests.

Done. Updated https://review.gluster.org/#/c/19905

**Comment by prashanthpai** [comment]
The CI tests are failing because the latest glusterfs nightly RPM contains this change: https://review.gluster.org/#/c/19734/ which broke starting of glusterd2.

**Comment by prashanthpai** [comment]
retest this please

**Pull request merged**